koswat/cost_report/summary/koswat_summary_location_matrix_builder.py

```diff
@@ -108,21 +89,46 @@ def get_reinforcement(
                 ground_level_surface=_reinforcement.new_ground_level_surface,
             )
 
+        _reinforcement_costs = [
+            next(iter(_reinforcement_dict.values()), None)
+            for _reinforcement_dict in locations_per_reinforcement
+        ]
+
+        return list(
+            map(
+                get_reinforcement,
+                filter(lambda x: x is not None, _reinforcement_costs),
+            )
+        )
```

Comment:
This looks more complicated than required. Do you want just the first value from the reinforcement dictionary? you can get it with `list`, otherwise I do not fully understand why you also use `next`.

Reply:
Refactored